### PR TITLE
PLAT-5519 Support for GCP deployment

### PR DESCRIPTION
**Pull request by sgmagar**
Ticket: https://crowdbotics.atlassian.net/browse/PLAT-5519
Updates to scaffold for GCP deployment support.

### CODE REVIEW

File: {{cookiecutter.project_slug}}/.circleci/generate_gcp_config.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      - run:
+          name: Webhook Failed
+          command: bash "$BACKEND_PATH.circleci/webhook_callback.sh" "failure"
+          when: on_fail
+
+workflows:
+  build_and_deploy_to_managed_workflow:
+    jobs:
+      - build_and_deploy
+EOF
```
Comment (driverdan) [inline {{cookiecutter.project_slug}}/.circleci/generate_gcp_config.sh:51]: Why does this need to be generated at runtime?

File: {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/settings.py [hunk 2]
```diff
@@ -11,17 +11,35 @@
...
+from google.auth.exceptions import DefaultCredentialsError
+from google.api_core.exceptions import PermissionDenied
 from modules.manifest import get_modules
 
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+env_file = os.path.join(BASE_DIR, ".env")
 env = environ.Env()
+env.read_env(env_file)
```
Comment (driverdan) [inline {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/settings.py:28]: Why is this necessary? `environ.Env()` should load `.env` automatically if it exists.

**Review by driverdan — CHANGES_REQUESTED** [review]

File: {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/settings.py (hunk 2 above)
Comment (sgmagar) [reply {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/settings.py:28]: `environ,Env()` this doesn't load .env file, we have to call read_env to load from file. In our case it's because of  `pipenv`, `pipenv shell` this loads if there is .env file.

File: {{cookiecutter.project_slug}}/.circleci/generate_gcp_config.sh (hunk 1 above)
Comment (sgmagar) [reply {{cookiecutter.project_slug}}/.circleci/generate_gcp_config.sh:51]: Well, if it was just django scaffold we wouldn't have to generate this at runtime. But in our case, this scaffold is used for mobile app alongside reactnative scaffold. So, now there will be 2 config file, one for mobile app and another for gcp. So, because of single repo and two config files, we conditionally generate config during build time.

File: {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/settings.py (hunk 2 above)
Comment (driverdan) [reply {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/settings.py:28]: You're right. I had checked the docs and misread them.

**Review by driverdan — APPROVED** [review]
